8.x/theme-development/helpers.md

```diff
@@ -10,424 +10,1608 @@ First, open the helper file: `app/Utils/CMS/Template/helpers.php`.
+__________________________
+
 
 
-### is_paste_possible()
+### is_paste_possible
+
 This function is for the admin panel and will be explained in the relevant section.
+__________________________
+
+
 
+### get_product_color_models
 
-### get_product_color_models()
 This function takes the product from the input and returns its colors.
+__________________________
+
+
 
+### get_product_last_color
 
-### get_product_last_color()
 Sorts and returns the last color for the product.
+__________________________
 
 
-### get_product_accessories()
+
+### get_product_accessories
+
 This function takes the product and returns the accessories of that product.
+__________________________
+
 
 
-### get_product_related_articles()
+### get_product_related_articles
+
 Returns articles related to a product to the number taken from the input.
+__________________________
+
+
 
+### get_product_related_products
 
-### get_product_related_products()
 Returns products related to a product to the specified number.
+__________________________
 
 
-### get_product_similar_products()
+
+### get_product_similar_products
+
 The get_product_similar_products function declares a similarity key for a product. For example, a car with the BMW brand is considered. This function returns products with the exact specifications as this car but with a different brand.
+__________________________
+
+
+
+
+### get_related_products_with_directory_level
 
-### get_related_products_with_directory_level()
 Returns related products having the same parent in which they are in specific higher levels passed by $level.
+__________________________
+
+
 
+### get_product_most_privileged_key_attributes
 
-### get_product_most_privileged_key_attributes()
 This function takes a list of all the keys and then returns the validity of that key. This function will be deprecated in the next version.
+__________________________
 
 
-### get_article_related_products()
+
+### get_article_related_products
+
 This function takes from the input of an article and returns related products.
+__________________________
+
 
 
-### get_article_related_articles()
+### get_article_related_articles
+
 This function takes from the input of an article and returns related articles.
+__________________________
+
+
 
+### recaptcha_enabled
 
-### recaptcha_enabled()
 This function determines whether Recaptcha is enabled on the system or not.
+__________________________
+
+
 
+### get_same_models_products
 
-### get_same_models_products()
 Takes a product from the input and returns same model products. For example: Different sizes of a shoe are same model products.
+__________________________
+
+
+
+### check_cart
 
-### check_cart()
 Checks if the product is in the cart. It is suggested not to use this function due to low efficiency.
+__________________________
+
+
 
+### get_cart_information
 
-### get_cart_information()
 This function returns cartRows data related to a product_id.
```

Comment:
This function returns cartRows data related to a product_id. -> This function returns `cartRows` data related to a `product_id`.